src/servers/launcher/launcher-review-request.md

```diff
@@ -0,0 +1,82 @@
+# Description
+
+For fun, and to learn some C++, I've decided to reimplement the entirety of EVE online as I remember it.  (I'm aware this is an insane and likely unreachable goal, that's intentional).
+
+I've started on the server side of things as I'm a little more comfortable with that than graphics programming, and settled on a cluster design, with one main `cluster` application handling startup and shutdown of the universe, one `launcher` application per server, one `station` server per space station, one `grid` server per simulated space instance, and so forth.
+
+Below is the _somewhat_ finished `launcher`.  It starts up, asks the database for its own IP address, and updates the database every second-ish as a sort of health check.  At the same time, it opens a UDP socket, and listens for messages.
+
+The messages start with their version, then the message type, then the body of that message (differing based on type), and ending with a sentinel value of `0x6C`.
+
+Two types of message are valid, one to launch instances, and one to shut down instances.  I've not written any of the other servers yet, so how to shut them down is as yet undiscovered, and the shutdown message in `launcher` is left unimplemented.
+
+`launcher` caps itself off by listening for `SIGTERM` or `SIGINT`, and gracefully shutting down when such a signal arrives.
+
+I've avoided using exceptions, except in constructors, where they're used to gracefully tear down an object from within an initialiser list.  This appears to be required for RAII, but I really don't like exceptions - I don't like their ability to bubble up and kill the server.  There's little worse than playing a video game and having it _hard_ crash to the desktop because some exception deep in the bowels of a particle system that the player doesn't really care about that much sneaks by.
+
+To that end, all constructors that throw are private, and the objects are constructed with `std::expected<T, E> T::tryCreate()`.
+
+# Things I plan to do at some point and maybe don't need to be told about
+
+* There's a lambda being used to `cout` a `variant` of exceptions in `main()` which may be able to be replaced with `cout`ing the `variant` directly now I implemented `operator<<` for `variant`s.
```

Comment:
No, no, no. Don’t do that. Never hijack the interface of other types.

`std::variant` does *not* belong to you, it belongs to the standard library. It is up to the standard library to decide if and how to format variants. What you are doing is opening the door to chaos. What happens if the standard library decides to add support for streaming variants in the future? And even if not, what happens if someone else, in some other library you happen to be using, decides they have the right to add a streaming operator for `std::variant`?

The correct thing to do is make your own type that *contains* the variant, and make *that* streamable. However, given what you are *actually* trying to do in this code, the actual correct thing to do here is: just don’t.

Reply:
> What happens if the standard library decides to add support for streaming variants in the future?

My compiler will tell me and I'll have to think of something else.

> what happens if someone else, in some other library you happen to be using, decides they have the right to add a streaming operator for std::variant?

Then I assume that's a terrible library and I probably shouldn't be using it.  The difference being that it's a library, not an executable.

I'm writing the end-user.  Nothing imports my code, it's not a library, if I want to add an overload to more easily print some standard lib types, I don't see a good reason why not.